_posts/2016/2016-05-31-gradle-jenkins-plugin.md

```diff
@@ -0,0 +1,112 @@
+We are using a [Gradle Jenkins plugin](https://github.com/ghale/gradle-jenkins-plugin/wiki) to control Jenkins installation.
+
+We already mention about the dsl, in the [Job DSL Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Job+DSL+Plugin)
+
+So you may ask why we changed it?
+Well, the other plugin you need to create a seed job, and at ePages, we have quite complex structure with a few shell scripts.
+Some of those shell scripts are shared on the company level, so before changing them, we want to make sure they work properly. (testing fase!)
+
+Another advantage of this gradle plugin is that is simple to deploy. For instance I didn't want to break our Team Jenkins installation, so I run a instance of Jenkins in a docker container and deploy my playground scripts to it.
+
+## How to setup the project
+In your main Project you should use the [gradle-jenkins-plugin](https://github.com/ghale/gradle-jenkins-plugin/wiki).
+
+I created a sample snippet for writing this blog post:
+{% highlight groovy %}
+buildscript {
+  repositories {
+    maven { url "https://plugins.gradle.org/m2/" }
+    maven { url 'http://repo.jenkins-ci.org/releases/' }
+    mavenLocal()
+  }
+  dependencies {
+    classpath "com.terrafolio:gradle-jenkins-plugin:1.3.2"
+  }
+}
+
+apply plugin: 'com.terrafolio.jenkins'
+
+jenkins {
+
+    servers {
+        localhost_jenkins {
+            url 'http://localhost:8080/' //change for your jenkins name
+            secure true
+            username 'admin' //change for your user name
+            password 'admin' //change for your password
+        }
+    }
+
+    defaultServer servers.localhost_jenkins
+
+    dsl {
+        job("SampleJob") {
+            displayName 'My sample job'
+            description ''
+            logRotator {
+                numToKeep(10)
+            }
+        }
+
+    } // end dsl
+}
+{% endhighlight %}
+
+## How to use
+
+I was using on my local Jenkins. After doing the proper changes on the dsl (jenkins.gradle), I just need to trigger the task updateJenkinsItems.
+
+To use it, you should create a gradle project and copy the snippet.
+On the project folder you could use:
+
+-----------------------
+{% highlight bash %}
+./gradlew tasks
+{% endhighlight %}
+
+This will provide you the list of available tasks:
+{% image blog/blog-gradlew_tasks.jpg %}
+
+-----------------------
+{% highlight bash %}
+./gradlew validateJenkinsItems
+{% endhighlight %}
+This will validate local changes with your Jenkins
+
+-----------------------
+{% highlight bash %}
+./gradlew updateJenkinsItems
+{% endhighlight %}
+This will update your Jenkins installation with the dsl defined at your jenkins.gradle
+{% image blog/blog-gradlew_updateJenkinsItems.jpg %}
+
+
+## Conclusions
+This plugin is really powerful and there are many cases where you could use to empower your jobs/automation environment. I hope the blog post ilustrate the applications of this plugin.
```

Comment:
This plugin is really powerful and there are many cases where you could use to empower your jobs/automation environment. I hope the blog post ilustrate the applications of this plugin.
>
This plugin is really powerful, and there are many cases where you could use it to empower your jobs/automation environment. I hope the blog post illustrates the application examples of this plugin.